include/cucumber-cpp/internal/CukeDll.hpp

```diff
@@ -0,0 +1,14 @@
+#ifndef CUKE_CUKEDLL_HPP_
```

Comment:
I'd appreciate it if you could use CMake's [GenerateExportHeader](https://cmake.org/cmake/help/v3.0/module/GenerateExportHeader.html) module, or at least make this header match the output that would be generated by that module much closer. To produce this output to get an example, add this to `src/CMakeLists.txt` and look for the generated files in `$BUILDDIR/src/`:
```cmake
include(GenerateExportHeader)

generate_export_header(cucumber-cpp-nomain)
generate_export_header(cucumber-cpp)
```

That should pave the way for symbol visibility on non-Windows OSes too, without having to redo the Windows part later on.

Additionally: to reflect that, please rename this header to not mention DLLs, but the fact that you're declaring which symbols are to be exported. E.g. `CukeExport.h`.